### PR TITLE
makefile: error if TESTS is set with make bench

### PR DESCRIPTION
**Pull request by RaduBerinde**

### CODE REVIEW

**Review by knz — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -24,8 +24,21 @@
 
 PKG          := ./pkg/...## Which package to run tests against, e.g. "./pkg/storage".
 TAGS         :=
-TESTS        :=.## Tests to run for use with `make test`.
-BENCHES      :=## Benchmarks to run for use with `make bench`.
+
+ifeq ($(MAKECMDGOALS),bench)
```
Comment (benesch) [inline Makefile]: This won't catch cases like `make build bench`. Can you use `ifneq ($(findstring bench,$(MAKECMDGOALS))),)` instead?

File: Makefile [hunk 2]
```diff
@@ -24,8 +24,21 @@
...
+ifneq ($(TESTS),)
+$(error TESTS specified with "make bench"; did you mean BENCHES?)
+endif
+TESTS := -
+BENCHES :=. ## Benchmarks to run for use with `make bench`.
```
Comment (benesch) [inline Makefile]: nit: spacing here looks funny.

**Review by benesch — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (RaduBerinde) [reply Makefile]: Done

File: Makefile (hunk 2 above)
Comment (RaduBerinde) [reply Makefile]: Fixed

**Comment by benesch** [comment]
🎉 LGTM! Thanks, Radu!

**Comment by RaduBerinde** [comment]
Actually, `ifneq ($(findstring bench,$(MAKECMDGOALS))),)` seems to always pass regardless of the target

**Comment by RaduBerinde** [comment]
Ah, there was an extra paren.

**Pull request merged**